custom/src/main/java/com/solarwinds/opentelemetry/extensions/initialize/config/HttpSettingsReader.java

```diff
@@ -0,0 +1,59 @@
+package com.solarwinds.opentelemetry.extensions.initialize.config;
+
+import com.solarwinds.joboe.config.ConfigManager;
+import com.solarwinds.joboe.config.ConfigProperty;
+import com.solarwinds.joboe.config.ServiceKeyUtils;
+import com.solarwinds.joboe.core.settings.SettingsReader;
+import com.solarwinds.joboe.logging.Logger;
+import com.solarwinds.joboe.logging.LoggerFactory;
+import com.solarwinds.joboe.sampling.Settings;
+import com.solarwinds.opentelemetry.extensions.ResourceCustomizer;
+import io.opentelemetry.semconv.incubating.HostIncubatingAttributes;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class HttpSettingsReader implements SettingsReader {
+
+  private final Pattern regex = Pattern.compile("https?(.*)");
```

Comment:
```suggestion

  private final Pattern regex = Pattern.compile("^https?:(.*)");
```
To avoid matching it in random parts of the URL by accident

Reply:
very unlikely. good callout though 😄